src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -39,6 +39,16 @@ private sealed partial class UIThreadOnly : IDisposable
 
             private readonly IInteractiveWindowEditorFactoryService _factory;
 
+            private readonly ITextBufferUndoManagerProvider _textBufferUndoManagerProvider;
+            private ITextBufferUndoManager UndoManager { set; get; }
+            public ITextUndoHistory UndoHistory
```

Comment:
What is ```UIThreadOnly``` exactly? If we're trying to ensure this is only used on the ui thread, then keep this private and have ```Undo(int)``` and ```Redo(int)``` here that do thread checks (see ForegroundThreadAffinitizedObject) and don't expose any *undomanager stuff at all.

(But maybe I don't understand the point of ```UIThreadOnly```)